ably/realtime_conn.go

```diff
@@ -224,20 +231,70 @@ func (c *Conn) State() StateEnum {
 // If no states are given, c is registered for all of them.
 // If c is nil, the method panics.
 // If c is already registered, its state set is expanded.
-func (c *Conn) On(ch chan<- State, states ...StateEnum) {
+func (c *Connection) On(ch chan<- State, states ...StateEnum) {
 	c.state.on(ch, states...)
 }
 
+// OnV12 registers an event handler for connection events of a specific kind.
+//
+// See package-level documentation on Event Emitter for details.
+func (c *Connection) OnV12(e ConnectionEventV12, handle func(ConnectionStateChangeV12)) (off func()) {
```

Comment:
I've left a couple of comments so far about the `V12` suffixes, are these just left over that need to be cleaned up or are we keeping some of them?

Reply:
Done at 2802c07bc5d05d240f2aa15af002b1fb4c9c4854, 98425261e9e04bad21faa49d1757a7377698f340.